standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestPartitionStat.java

```diff
@@ -0,0 +1,242 @@
+import org.apache.hadoop.hive.metastore.client.builder.DatabaseBuilder;
+import org.apache.hadoop.hive.metastore.client.builder.PartitionBuilder;
+import org.apache.hadoop.hive.metastore.client.builder.TableBuilder;
+import org.apache.hadoop.hive.metastore.columnstats.cache.LongColumnStatsDataInspector;
+import org.apache.hadoop.hive.metastore.conf.MetastoreConf;
+import org.apache.hadoop.hive.metastore.minihms.AbstractMetaStoreService;
+import org.apache.hadoop.hive.metastore.utils.FileUtils;
+
+import org.apache.thrift.TException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Tests for updating partition column stats.
+ */
+@RunWith(Parameterized.class)
+@Category(MetastoreCheckinTest.class)
+public class TestPartitionStat extends MetaStoreClientTest {
+  private AbstractMetaStoreService metaStore;
+  private IMetaStoreClient client;
+  private String directSql = "false";
+
+  protected static final String DB_NAME = "test_part_stat";
+  protected static final String TABLE_NAME = "test_part_stat_table";
+  private static final String DEFAULT_COL_TYPE = "int";
+  private static final String PART_COL_NAME = "year";
+  protected static final short MAX = -1;
+  private static final Partition[] PARTITIONS = new Partition[5];
+  public static final String HIVE_ENGINE = "hive";
+
+  public TestPartitionStat(String name, AbstractMetaStoreService metaStore) {
+    this.metaStore = metaStore;
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    // Get new client, store the original value of directSql to restore it back after test.
+    directSql = metaStore.getConf().get(MetastoreConf.ConfVars.TRY_DIRECT_SQL.getVarname());
+    client = metaStore.getClient();
+
+    // Clean up the database
+    client.dropDatabase(DB_NAME, true, true, true);
+    metaStore.cleanWarehouseDirs();
+    Database db = new DatabaseBuilder().
+        setName(DB_NAME).
+        create(client, metaStore.getConf());
+
+    // Create test tables with 3 partitions
+    createTable(TABLE_NAME, getYearPartCol());
+    createPartitions();
+
+    // Hack to initialize the stats tables. Not sure why the first test run using remote metastore is failing with
+    // error (Unable to update Column stats for  table due to: Table/View 'PART_COL_STATS' does not exist.). This hack
+    // has fixed it. All the tests will be executed first using embedded metastore and then with
+    // remote metastore. For embedded metastore, TRY_DIRECT_SQL is set to false. For remote metastore TRY_DIRECT_SQL is
+    // set to true.
+    client.getPartitionColumnStatistics(DB_NAME, TABLE_NAME,
```

Comment:
My guess is that the autocreate scripts do not create the tables just when there is a DN request for it. Since we are using directSql to get the data the tables are not created.

I would guess that we have to create the appropriate tables from `MetaStoreDirectSql.ensureDbInit()`